dashboard/src/actions/profileActions.js

```diff
@@ -0,0 +1,92 @@
+import * as TYPES from "./types";
+import API from "../utils/axiosInstance";
+import * as API_ROUTES from "../utils/apiConstants";
+import Cookies from "js-cookie";
+
+export const getProfileDetails = () => async (dispatch) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const username = Cookies.get("username") ? Cookies.get("username") : null;
+
+    const response = await API.get(`${API_ROUTES.GET_USER}/${username}`);
+
+    if (response.status === 200 && Object.keys(response.data).length > 0) {
+      dispatch({
+        type: TYPES.GET_USER_DETAILS,
+        payload: response?.data,
+      });
+    } else {
+      let toast = {
+        variant: "danger",
+        title: response?.data?.message,
+      };
+      dispatch({
+        type: TYPES.SHOW_TOAST,
+        payload: { ...toast },
+      });
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  } catch (error) {
+    dispatch({ type: TYPES.NETWORK_ERROR });
+    dispatch({ type: TYPES.COMPLETED });
+  }
+};
+
+export const updateUserDetails =
+  (value, fieldName) => async (dispatch, getState) => {
+    let userDetails = { ...getState().userProfile.userDetails };
+    userDetails[fieldName] = value;
+
+    dispatch({
+      type: TYPES.UPDATE_USER_DETAILS,
+      payload: { ...userDetails },
+    });
+  };
+
+export const sendForUpdation = () => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const userDetails = { ...getState().userProfile.userDetails };
+    const userDetails_copy = { ...getState().userProfile.userDetails_copy };
```

Comment:
Can we consider inverting the names?  Instead of using `_copy` to mark the values that are the actual, perhaps we could consider `userDetails` to be the original copy, and add `userDetails_updates` where it is empty until somebody makes changes, and then when the final button is pressed to commit them, the values are moved from `_updates` to `userDetails`?

Reply:
The `_copy` is needed for the `reset` functionality. Because only with `userDetails` the UI is rendered. On changing the input values and for that to reflect in the UI, need to update the `userDetails` object.

Code changes:
-  Modified the code with `_updates` where only the updated fields get inside it and sent for update
-  `Save` button is disabled until the user updates any of the fields
-  `Cancel` and `Reset to default` both the buttons are maintained. If not, the `Settings` card would have lot of empty space

**Working of `Cancel` button**
Reverts back to the `normal` view from the `edit` view while retaining the last saved state

**Working of `Reset` button**
Reverts to the last saved state but maintains the `edit` view